src/AuthenticationApp/MainComponents/SignUpPage.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from "react";
+import { Link, navigate } from "@reach/router";
+import { authentication, generateUserDocument } from "../firebase";
+ 
+const SignUpPage = () => {
+    const [userEmail, setUserEmail] = useState("");
+    const [userPassword, setUserPassword] = useState("");
+    const [retypedPassword, setRetypedPassword] = useState("");
+    const [displayName, setDisplayName] = useState("");
+    const [error, setError] = useState(null);
+ 
+    const createUser = async (
+        event, 
+        userEmail, 
+        userPassword, 
+        displayName,
+        retypedPassword
+        ) => {
+            event.preventDefault();
+            
+            try {
+                if (retypedPassword !== userPassword) {
+                    throw new SyntaxError("Your passwords don't match");
+                } else {  
+                    const {user} = await authentication.createUserWithEmailAndPassword(
+                        userEmail, 
+                        userPassword
+                        );
+                    user.updateProfile({
+                        displayName: displayName
```

Comment:
Aren't we doing any validations on displayName and email to be unique?
What about if user enters non edu email id?